src/services/giant_api.rs

```diff
@@ -132,3 +132,38 @@ pub fn get_blobs_in_collection(uri: &str, collection: &str) -> Result<Vec<Blob>,
         Err(CliError::UnexpectedResponse(status))
     }
 }
+
+pub fn delete_blob(giant_uri: &str, blob_uri: &str) -> Result<(), CliError> {
+    // TODO QUESTION: is it wasteful to keep getting this client over and over?
+    let client = get_client(giant_uri)?;
```

Comment:
A `reqwest` client contains a connection pool so it's not perfect to create a client in a hot loop.

You can pull the client into a struct and move these to level functions into member functions which will allow easy reuse.

Reply:
What Sam said - alternatively, if you don't want to bother with a full client struct, you can change the signature of your `delete_blob` and `delete_collection` function to take a `client: &reqwest::Client` as an extra parameter, so each invocation in the for loop with reference the existing client instead of newing one up.

If you run into ownership issues inside the for loop (I don't think you will, but just in case), you can also safely call `.clone()` on the client and it won't new up a new one, since`reqwest::Client` maintains an internal pool of connections with an atomic reference counter`[Arc](https://doc.rust-lang.org/std/sync/struct.Arc.html)`, meaning you can call `.clone()` as many times as you want and it will still reuse the existing pool.

Reply:
👍 great advice, thanks

Reply:
> move these to level functions into member functions

sorry, not quite sure how to parse this - what's a "level function" or a "to level function"?

and when you say "member functions" I guess you mean functions that are members of a `struct`. Is this the same as a method, or associated function? as per https://doc.rust-lang.org/rust-by-example/fn/methods.html

Reply:
Ah sorry should have been "top level function", aka just a loose function in a module.

Yeah member function like in Java, so in Rust world a function that lives in an `impl` block and takes some type of `self` as an argument, in this case probably `&self`

Reply:
Yeah I like the idea of having a proper `GiantApiClient` struct but I'll do in a separate PR